packages/engine/src/script/ScriptComponent.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/ir-engine/ir-engine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import { cleanStorageProviderURLs } from '@ir-engine/common/src/utils/parseSceneJSON'
+import { defineComponent, useComponent, useEntityContext } from '@ir-engine/ecs'
+import { useEffect } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+
+export const ScriptComponent = defineComponent({
+  name: 'ScriptComponent',
+  jsonID: 'EE_script',
+
+  onInit: (entity) => {
+    return {
+      uuid: '',
```

Comment:
we only need `src` here, the other options are unnecessary

Reply:
roger that, how do we prevent the script from mounting and executing in the editor the moment we append to dom
we could just skip appending to dom unless we are in location or play mode, that a solution

Reply:
made the necessary changes